opendeplete/openmc_wrapper.py

```diff
@@ -572,99 +456,71 @@ def initial_condition(self):
             Total density for initial conditions.
         """
 
-        # Write geometry.xml
-        if self.rank == 0:
+        # Create XML files
+        if comm.rank == 0:
             self.geometry.export_to_xml()
+            self.generate_settings_xml()
+            self.generate_materials_xml()
+        self.generate_tally_xml()
+
+        # Initialize OpenMC library
+        openmc.capi.init(comm)
 
         # Return number density vector
         return self.total_density_list()
 
+    def _update_materials(self):
+        """Updates material compositions in OpenMC on all processes."""
+
+        number_list = comm.allgather(self.number)
+
+        for number_i in number_list:
+            for mat in number_i.mat_to_ind:
+                nuclides = []
+                densities = []
+                for nuc in self.number.nuc_to_ind:
+                    if nuc in self.participating_nuclides:
+                        val = 1.0e-24*self.number.get_atom_density(mat, nuc)
+
+                        # If nuclide is zero, do not add to the problem.
+                        if val > 0.0:
+                            if self.settings.round_number:
+                                val_magnitude = np.floor(np.log10(val))
+                                val_scaled = val / 10**val_magnitude
+                                val_round = round(val_scaled, 8)
+
+                                val = val_round * 10**val_magnitude
+
+                            nuclides.append(nuc)
+                            densities.append(val)
+                        else:
+                            # Only output warnings if values are significantly
+                            # negative.  CRAM does not guarantee positive values.
+                            if val < -1.0e-21:
+                                print("WARNING: nuclide ", nuc, " in material ", mat,
+                                      " is negative (density = ", val, " at/barn-cm)")
+                            number_i[mat, nuc] = 0.0
+
+                mat_view = openmc.capi.materials[int(mat)]
+                mat_view.set_densities(nuclides, densities)
+
     def generate_materials_xml(self):
         """ Creates materials.xml from self.number.
 
         Due to uncertainty with how MPI interacts with OpenMC API, this
```

Comment:
I would update the rest of this comment, after a path is chosen for how to write the initial condition.